configuretools.cmake

```diff
@@ -0,0 +1,75 @@
+# Compiler tools only have the major and minor version in their name
+if(CMAKE_CXX_COMPILER_VERSION MATCHES "^([0-9]+).([0-9]+)")
+  set(CLR_CMAKE_COMPILER_TOOL_VERSION "${CMAKE_MATCH_1}.${CMAKE_MATCH_2}")
+else()
+  set(CLR_CMAKE_COMPILER_TOOL_VERSION ${CMAKE_CXX_COMPILER_VERSION})
+endif()
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  if(APPLE)
+    set(LLVM_PREFIX "")
+  else()
+    set(LLVM_PREFIX "llvm-")
+  endif()
+  function(locate_llvm_exec exec var)
+    find_program(EXEC_LOCATION
+     NAMES "${LLVM_PREFIX}${exec}-${CLR_CMAKE_COMPILER_TOOL_VERSION}" "${LLVM_PREFIX}${exec}")
```

Comment:
This is different from what we were doing in the .sh files. Over there, we were using the version that was found to be attached to clang command. So, if we've found that there is just `clang`, we were not attaching any version when looking for the executables. Now we are always attaching the version number that cmake gives us.
Another case is that the version is expressed in different ways in the command name. So e.g. for clang 3.9, it can be clang-3.9, clang39 or clang-39, depending on the distro / OS.
I think we need to use the `desired_llvm_version` we've detected in the find-clang.sh here.

Reply:
I think if we do a matching fallback here with the different version syntaxes, then we'll always find the correct match. I'd prefer to not have to pass through another environment variable/CMake variable to the CMake script if possible.

Reply:
It seems to me that duplicating the same logic at two places, moreover in two different languages is a recipe for a future problem.
But we can actually parse it out of the CMAKE_C_COMPILER, so we don't need an extra variable for it.

Reply:
I'll change it to parse the version out of the name.